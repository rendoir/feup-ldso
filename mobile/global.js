global.api = '10.227.150.123';
global.userId = null;
global.userName = null;
