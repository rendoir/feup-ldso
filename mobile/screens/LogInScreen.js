import React from 'react';
import { Font, AppLoading } from "expo";
import { Root } from 'native-base';
import { Text, View, Image, Button, StatusBar } from "react-native";

export default class LogInScreen extends React.Component {

    constructor(props) {
        super(props);
        this.state = { signedIn: false, userName: "", userToken: "" };
    }

    static navigationOptions = {
        header: null
    };

    componentWillUnmount() {
        this.isCancelled = true;
    }

    async componentDidMount() {
        await Font.loadAsync({
            Roboto: require("native-base/Fonts/Roboto.ttf"),
            Roboto_medium: require("native-base/Fonts/Roboto_medium.ttf")
        });
        this.setState({ loading: false });
    }

    render() {
        if (this.state.loading) {
            return (
                <Root>
                    <AppLoading />
                </Root>
            );
        }

        return (
            <View style={{ backgroundColor: '#2c8f7f', height: '100%' }}>
                <StatusBar hidden />

                <View
                    style={{
                        position: 'absolute',
                        top: 0,
                        left: 0,
                        width: '100%',
                        height: '100%'
                    }}>
                    <Image
                        style={{
                            flex: 1,
                            width: '100%',
                            height: '100%',
                            opacity: 0.7
                        }}
                        source={{ uri: "https://upload.wikimedia.org/wikipedia/commons/4/4a/Edif%C3%ADcio_da_Reitoria_da_Universidade_do_Porto.jpg" }}
                    />
                </View>

                <View style={{ alignItems: 'center', flexDirection: 'column', height: '99%' }}>
                    <View style={{ flex: 3 }}></View>
                    <View style={{ flex: 1 }}>
                        <Image source={require('../assets/images/original.png')}
                            style={{
                            }}
                        />
                    </View>
                    <View style={{ flex: 1 }}>
                        <Text style={{ fontSize: 25, color: 'white', fontFamily: 'Roboto', textAlign: 'center', marginHorizontal: '10%' }} >
                            Agenda Cultural da Universidade do Porto
                        </Text>
                    </View>
                    <View style={{ width: '50%', flex: 1 }}>
                        <Button title="Sign in with Google" className={'login_button'} onPress={() => this.props.signIn()} />
                    </View>
                    <View style={{ flex: 2 }}>{this.props.signInErrorMessage}</View>
                </View>
            </View>

        );
    }
}
