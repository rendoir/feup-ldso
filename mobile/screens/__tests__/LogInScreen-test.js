/**
* @jest-environment jsdom
*/
import 'react-native';
import React from 'react';
import LogInScreen from '../LogInScreen';
import renderer from 'react-test-renderer';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';
import { shallow } from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import '../../global.js';

Enzyme.configure({ adapter: new Adapter() });

describe('App snapshot', () => {
    jest.useFakeTimers();
    beforeEach(() => {
        NavigationTestUtils.resetInternalState();
        const global = jest.genMockFromModule('../../global.js');
        global.dictionary = jest.fn(() => "");
    });

    it('renders correctly', async() => {
        const wrapper = shallow(<LogInScreen />);
        expect(wrapper).toMatchSnapshot();
    });

    it('renders loading', async() => {
        const wrapper = shallow(<LogInScreen />);
        wrapper.setState({ loading: true });
        expect(wrapper).toMatchSnapshot();
    });

    it('mounts correctly', () => {
        const tree = shallow(<LogInScreen />);
        setImmediate(() => {
            tree.componentDidMount();
            expect(tree.state.isCancelled).toEqual(false);
            expect(tree.state.loading).toEqual(false);
        });
    });

    it('unmounts correctly', () => {
        const tree = renderer.create(<LogInScreen language={'PT'}/>).getInstance();
        tree.componentWillUnmount();
        expect(tree.isCancelled).toEqual(true);
    });

    it('navigates correctly', () => {
        let signIn = jest.fn();
        const tree = shallow(<LogInScreen language={'PT'} signIn={signIn}/>);

        let button = tree.find(".login_button").first();
        button.props().onPress();

        expect(signIn).toHaveBeenCalled();
    });

});
