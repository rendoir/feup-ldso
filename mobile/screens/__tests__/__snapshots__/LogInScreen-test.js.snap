// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LogInScreen />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <StatusBar
          animated={false}
          hidden={true}
          showHideTransition="fade"
        />,
        <Component
          style={
            Object {
              "height": "100%",
              "left": 0,
              "position": "absolute",
              "top": 0,
              "width": "100%",
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://upload.wikimedia.org/wikipedia/commons/4/4a/Edif%C3%ADcio_da_Reitoria_da_Universidade_do_Porto.jpg",
              }
            }
            style={
              Object {
                "flex": 1,
                "height": "100%",
                "opacity": 0.7,
                "width": "100%",
              }
            }
          />
        </Component>,
        <Component
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "column",
              "height": "99%",
            }
          }
        >
          <Component
            style={
              Object {
                "flex": 3,
              }
            }
          />
          <Component
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Image
              source={1}
              style={Object {}}
            />
          </Component>
          <Component
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "white",
                  "fontFamily": "Roboto",
                  "fontSize": 25,
                  "marginHorizontal": "10%",
                  "textAlign": "center",
                }
              }
            >
              Agenda Cultural da Universidade do Porto
            </Text>
          </Component>
          <Component
            style={
              Object {
                "flex": 1,
                "width": "50%",
              }
            }
          >
            <Button
              className="login_button"
              onPress={[Function]}
              title="Sign in with Google"
            />
          </Component>
          <Component
            style={
              Object {
                "flex": 2,
              }
            }
          />
        </Component>,
      ],
      "style": Object {
        "backgroundColor": "#2c8f7f",
        "height": "100%",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "animated": false,
          "hidden": true,
          "showHideTransition": "fade",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Image
            source={
              Object {
                "uri": "https://upload.wikimedia.org/wikipedia/commons/4/4a/Edif%C3%ADcio_da_Reitoria_da_Universidade_do_Porto.jpg",
              }
            }
            style={
              Object {
                "flex": 1,
                "height": "100%",
                "opacity": 0.7,
                "width": "100%",
              }
            }
          />,
          "style": Object {
            "height": "100%",
            "left": 0,
            "position": "absolute",
            "top": 0,
            "width": "100%",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "source": Object {
              "uri": "https://upload.wikimedia.org/wikipedia/commons/4/4a/Edif%C3%ADcio_da_Reitoria_da_Universidade_do_Porto.jpg",
            },
            "style": Object {
              "flex": 1,
              "height": "100%",
              "opacity": 0.7,
              "width": "100%",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Component
              style={
                Object {
                  "flex": 3,
                }
              }
            />,
            <Component
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <Image
                source={1}
                style={Object {}}
              />
            </Component>,
            <Component
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "white",
                    "fontFamily": "Roboto",
                    "fontSize": 25,
                    "marginHorizontal": "10%",
                    "textAlign": "center",
                  }
                }
              >
                Agenda Cultural da Universidade do Porto
              </Text>
            </Component>,
            <Component
              style={
                Object {
                  "flex": 1,
                  "width": "50%",
                }
              }
            >
              <Button
                className="login_button"
                onPress={[Function]}
                title="Sign in with Google"
              />
            </Component>,
            <Component
              style={
                Object {
                  "flex": 2,
                }
              }
            />,
          ],
          "style": Object {
            "alignItems": "center",
            "flexDirection": "column",
            "height": "99%",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "style": Object {
                "flex": 3,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Image
                source={1}
                style={Object {}}
              />,
              "style": Object {
                "flex": 1,
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "source": 1,
                "style": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "white",
                    "fontFamily": "Roboto",
                    "fontSize": 25,
                    "marginHorizontal": "10%",
                    "textAlign": "center",
                  }
                }
              >
                Agenda Cultural da Universidade do Porto
              </Text>,
              "style": Object {
                "flex": 1,
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "Agenda Cultural da Universidade do Porto",
                "ellipsizeMode": "tail",
                "style": Object {
                  "color": "white",
                  "fontFamily": "Roboto",
                  "fontSize": 25,
                  "marginHorizontal": "10%",
                  "textAlign": "center",
                },
              },
              "ref": null,
              "rendered": "Agenda Cultural da Universidade do Porto",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Button
                className="login_button"
                onPress={[Function]}
                title="Sign in with Google"
              />,
              "style": Object {
                "flex": 1,
                "width": "50%",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "login_button",
                "onPress": [Function],
                "title": "Sign in with Google",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": undefined,
              "style": Object {
                "flex": 2,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <StatusBar
            animated={false}
            hidden={true}
            showHideTransition="fade"
          />,
          <Component
            style={
              Object {
                "height": "100%",
                "left": 0,
                "position": "absolute",
                "top": 0,
                "width": "100%",
              }
            }
          >
            <Image
              source={
                Object {
                  "uri": "https://upload.wikimedia.org/wikipedia/commons/4/4a/Edif%C3%ADcio_da_Reitoria_da_Universidade_do_Porto.jpg",
                }
              }
              style={
                Object {
                  "flex": 1,
                  "height": "100%",
                  "opacity": 0.7,
                  "width": "100%",
                }
              }
            />
          </Component>,
          <Component
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "column",
                "height": "99%",
              }
            }
          >
            <Component
              style={
                Object {
                  "flex": 3,
                }
              }
            />
            <Component
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <Image
                source={1}
                style={Object {}}
              />
            </Component>
            <Component
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "white",
                    "fontFamily": "Roboto",
                    "fontSize": 25,
                    "marginHorizontal": "10%",
                    "textAlign": "center",
                  }
                }
              >
                Agenda Cultural da Universidade do Porto
              </Text>
            </Component>
            <Component
              style={
                Object {
                  "flex": 1,
                  "width": "50%",
                }
              }
            >
              <Button
                className="login_button"
                onPress={[Function]}
                title="Sign in with Google"
              />
            </Component>
            <Component
              style={
                Object {
                  "flex": 2,
                }
              }
            />
          </Component>,
        ],
        "style": Object {
          "backgroundColor": "#2c8f7f",
          "height": "100%",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "animated": false,
            "hidden": true,
            "showHideTransition": "fade",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Image
              source={
                Object {
                  "uri": "https://upload.wikimedia.org/wikipedia/commons/4/4a/Edif%C3%ADcio_da_Reitoria_da_Universidade_do_Porto.jpg",
                }
              }
              style={
                Object {
                  "flex": 1,
                  "height": "100%",
                  "opacity": 0.7,
                  "width": "100%",
                }
              }
            />,
            "style": Object {
              "height": "100%",
              "left": 0,
              "position": "absolute",
              "top": 0,
              "width": "100%",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "source": Object {
                "uri": "https://upload.wikimedia.org/wikipedia/commons/4/4a/Edif%C3%ADcio_da_Reitoria_da_Universidade_do_Porto.jpg",
              },
              "style": Object {
                "flex": 1,
                "height": "100%",
                "opacity": 0.7,
                "width": "100%",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Component
                style={
                  Object {
                    "flex": 3,
                  }
                }
              />,
              <Component
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <Image
                  source={1}
                  style={Object {}}
                />
              </Component>,
              <Component
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "white",
                      "fontFamily": "Roboto",
                      "fontSize": 25,
                      "marginHorizontal": "10%",
                      "textAlign": "center",
                    }
                  }
                >
                  Agenda Cultural da Universidade do Porto
                </Text>
              </Component>,
              <Component
                style={
                  Object {
                    "flex": 1,
                    "width": "50%",
                  }
                }
              >
                <Button
                  className="login_button"
                  onPress={[Function]}
                  title="Sign in with Google"
                />
              </Component>,
              <Component
                style={
                  Object {
                    "flex": 2,
                  }
                }
              />,
            ],
            "style": Object {
              "alignItems": "center",
              "flexDirection": "column",
              "height": "99%",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "style": Object {
                  "flex": 3,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Image
                  source={1}
                  style={Object {}}
                />,
                "style": Object {
                  "flex": 1,
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "source": 1,
                  "style": Object {},
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "white",
                      "fontFamily": "Roboto",
                      "fontSize": 25,
                      "marginHorizontal": "10%",
                      "textAlign": "center",
                    }
                  }
                >
                  Agenda Cultural da Universidade do Porto
                </Text>,
                "style": Object {
                  "flex": 1,
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "accessible": true,
                  "allowFontScaling": true,
                  "children": "Agenda Cultural da Universidade do Porto",
                  "ellipsizeMode": "tail",
                  "style": Object {
                    "color": "white",
                    "fontFamily": "Roboto",
                    "fontSize": 25,
                    "marginHorizontal": "10%",
                    "textAlign": "center",
                  },
                },
                "ref": null,
                "rendered": "Agenda Cultural da Universidade do Porto",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Button
                  className="login_button"
                  onPress={[Function]}
                  title="Sign in with Google"
                />,
                "style": Object {
                  "flex": 1,
                  "width": "50%",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "className": "login_button",
                  "onPress": [Function],
                  "title": "Sign in with Google",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": undefined,
                "style": Object {
                  "flex": 2,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`App snapshot renders loading 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LogInScreen />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <AppLoading />,
      "virtual": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <AppLoading />,
        "virtual": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
